File: w10/d04/jungle_tree/app/controllers/monkeys_controller.rb
class MonkeysController < ApplicationController

# monkeys GET    /monkeys(.:format)     monkeys#index
def index
  monkeys = Monkey.all
  render json: monkeys.to_json
end

#         POST   /monkeys(.:format)     monkeys#create
def create
  monkey = Monkey.create(monkey_params)
  render json: monkey.to_json
end

#  monkey GET    /monkeys/:id(.:format) monkeys#show
def show
  monkey = Monkey.find(params[:id])
  render json: monkey.to_json
end

#         PATCH  /monkeys/:id(.:format) monkeys#update
#         PUT    /monkeys/:id(.:format) monkeys#update
def update
  monkey = Monkey.find(params[:id])
  monkey.update(monkey_params)
  render json: monkey.to_json
end

#         DELETE /monkeys/:id(.:format) monkeys#destroy
def destroy
  monkey = Monkey.find(params[:id])
  monkey.destroy
  render json: monkey.to_json
end

private

def monkey_params
  params.require(:monkey).permit(:name)
end

end

File: w11/d01/node_bananas/monkeysController.js
var Monkey = require(path.join(__dirname, 'monkey'));

exports.index = function(req, res){
  Monkey.find(function(err, monkeys){
    if (err) return console.log(err);
    res.send(monkeys);
  });
}

exports.create = function(req, res){
  Monkey.create({name: req.body.name}, function(err, mongo){
    if (err) return console.log(err);
    res.send(monkey);
  });
}

exports.update = function(req, res){
  Monkey.findById(req.params.id, function(err, monkey){
    if (err) return console.error(err);
    monkey.update({name: req.body.name}, function(err, update){
      if (err) return console.error(err);
      res.send(monkey)
    });
  })
}

exports.delete = function(req, res){
  Monkey.findById(req.params.id, function(err, monkey){
    if (err) return console.error(err);
    monkey.remove(function(err){
      if (err) return console.error(err);
      return res.send({'success' : 'ok'});
    })
  });
}
